### PR TITLE
Last bug fixked

### PR DESCRIPTION
**Pull request by ChristofferLeo**
It fails merge conflict_test as this has been manipulated to just test really fast.

### CODE REVIEW

**Comment by ChristofferLeo** [comment]
However I see it fails merge conflicts. This test has been manipulated to debug quickly. I.e coordinator was set to transmit the new elevator date after it had merged. In the test it is waiting for this, and will therefore have a timeout

File: config.toml [hunk 1]
```diff
@@ -16,6 +16,6 @@ motor_driving_timeout = 10000
 
 [hardware]
 n_floors = 4
-driver_address = "localhost"
-driver_port = 15657
+driver_address = "host.docker.internal"
```
Comment (Mathiasotnes) [inline config.toml:19]: This shouldn't be changed to simulator values, the code must run on the elevators.

File: src/coordinator/coordinator.rs [hunk 2]
```diff
@@ -408,7 +407,16 @@ impl Coordinator {
     }
 
     fn check_merge_type(&self, elevator_data: ElevatorData) -> MergeType {
-        let new_elevators = elevator_data.states.keys().len() > self.elevator_data.states.keys().len();
+        let mut new_elevators = false;
+        for key in self.elevator_data.states.keys() {
+            if elevator_data.states.contains_key(key) {
+                new_elevators = false;
+                print!("No new elevators \n");
```
Comment (Mathiasotnes) [inline src/coordinator/coordinator.rs:414]: Don't use prints, use logs.

File: src/coordinator/coordinator.rs [hunk 3]
```diff
@@ -408,7 +407,16 @@ impl Coordinator {
     }
 
     fn check_merge_type(&self, elevator_data: ElevatorData) -> MergeType {
-        let new_elevators = elevator_data.states.keys().len() > self.elevator_data.states.keys().len();
+        let mut new_elevators = false;
+        for key in self.elevator_data.states.keys() {
```
Comment (Mathiasotnes) [inline src/coordinator/coordinator.rs:411]: Is there a case for when an elevator isn't in the list? Should the merges be triggered whenever the states ID's differ? And not just if one contains more ID's than another? I'm not sure.

**Review by Mathiasotnes — CHANGES_REQUESTED** [review]

**Pull request merged**